src/libs/costmap_2d/CMakeLists.txt

```diff
@@ -1,202 +1,161 @@
+endif()
+
 
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(nav_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(rmw REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(map_msgs REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(xmlrpcpp REQUIRED)
+find_package(laser_geometry REQUIRED)
+find_package(pluginlib REQUIRED)
+
+#	TODO(bpwilcox): port remaining packages
+#find_package(tf2_sensor_msgs REQUIRED)
+#find_package(dynamic_reconfigure REQUIRED) 
+#find_package(message_filters REQUIRED)
+#find_package(voxel_grid REQUIRED)
+
+remove_definitions(-DDISABLE_LIBUSB-1.0)
 find_package(Eigen3 REQUIRED)
 find_package(Boost REQUIRED COMPONENTS system thread)
+
 include_directories(
-    include
-    ${catkin_INCLUDE_DIRS}
-    ${EIGEN3_INCLUDE_DIRS}
-    ${Boost_INCLUDE_DIRS}
+	include
+	${ament_INCLUDE_DIRS}
+	${EIGEN3_INCLUDE_DIRS}
+	${Boost_INCLUDE_DIRS}
 )
 
 add_definitions(${EIGEN3_DEFINITIONS})
 
-# messages
-add_message_files(
-    DIRECTORY msg
-    FILES
-    VoxelGrid.msg
-)
+#	TODO(bpwilcox): Separate msgs generation into separate package
+#rosidl_generate_interfaces(${PROJECT_NAME}
+#  "msg/VoxelGrid.msg"
+#
+#	DEPENDENCIES std_msgs geometry_msgs map_msgs
+#)
 
-generate_messages(
-    DEPENDENCIES
-        std_msgs
-        geometry_msgs
-        map_msgs
-)
 
-# dynamic reconfigure
-generate_dynamic_reconfigure_options(
-  cfg/Costmap2D.cfg
-  cfg/ObstaclePlugin.cfg
-  cfg/GenericPlugin.cfg
-  cfg/InflationPlugin.cfg
-  cfg/VoxelPlugin.cfg
-)
-
-catkin_package(
-    INCLUDE_DIRS
-        include
-        ${EIGEN3_INCLUDE_DIRS}
-    LIBRARIES costmap_2d layers
-    CATKIN_DEPENDS
-        dynamic_reconfigure
-        geometry_msgs
-        laser_geometry
-        map_msgs
-        message_filters
-        message_runtime
-        nav_msgs
-        pluginlib
-        roscpp
-        sensor_msgs
-        std_msgs
-        tf2_ros
-        visualization_msgs
-        voxel_grid
-    DEPENDS
-        EIGEN3
-        Boost
-)
-
-add_library(costmap_2d
+add_library(nav2_costmap_2d
   src/array_parser.cpp
   src/costmap_2d.cpp
-  src/observation_buffer.cpp
   src/layer.cpp
   src/layered_costmap.cpp
   src/costmap_2d_ros.cpp
   src/costmap_2d_publisher.cpp
   src/costmap_math.cpp
   src/footprint.cpp
-  src/costmap_layer.cpp
+  src/costmap_layer.cpp	
+  #	TODO(bpwilcox): need ROS PCL dependencies to port
+  #src/observation_buffer.cpp
+  
+)
+
+set(dependencies
+  rclcpp
+  geometry_msgs
+  nav_msgs
+  map_msgs
+  tf2_ros
+  tf2
+  tf2_sensor_msgs
+  xmlrpcpp
+  pluginlib
+  tf2_geometry_msgs
 )
-add_dependencies(costmap_2d ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(costmap_2d
+
+ament_target_dependencies(nav2_costmap_2d 
+${dependencies}
```

Comment:
Two space indentation (should be consistent with the rest of this and similar files).